x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotIndexInput.java

```diff
@@ -108,19 +197,24 @@ protected void seekInternal(long pos) throws IOException {
         } else if (pos < 0L) {
             throw new IOException("Seeking to negative position [" + pos + "] for " + toString());
         }
-        this.position = offset + pos;
+        if (position != offset + pos) {
+            position = offset + pos;
+            IOUtils.close(streamForSequentialReads);
+            streamForSequentialReads = null;
```

Comment:
Maybe nullify in a finally block, just in case

Reply:
Oh good point, done in c9cf7bcd11d91a676f60d3a6bdb4c9b44858c368.